src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -27,51 +31,63 @@ export type StreamNameParts<T extends StreamType = StreamType> = {
   streamId: string;
 };
 
-export type StreamToProject<EventType extends Event> = {
-  streamName: StreamName;
-  streamType: StreamType;
-  streamId: string;
-  streamVersion: number;
-  events: ReadEvent<EventType, ReadEventMetadata>[];
-};
-
-export type Projection<EventType extends Event> = (
-  stream: StreamToProject<EventType>,
-) => void | Promise<void>;
-
-export interface EventStream<EventType extends Event = Event> {
+export interface EventStream<
+  EventType extends Event = Event,
+  ShortInfoType extends DefaultRecord = DefaultRecord,
+> {
   streamName: string;
   events: Array<ReadEvent<EventType, ReadEventMetadata>>;
-  createdAt: Date;
-  updatedAt: Date;
+  metadata: {
+    streamId: string;
+    streamType: StreamType;
+    streamPosition: bigint;
+    createdAt: Date;
+    updatedAt: Date;
+  };
+  projection: {
```

Comment:
**FIX:** I think that it'd be better to make projection either a list of projections or a record. People can have more than two, or have just one, or none.

Reply:
Yep I've been thinking about that as I was working on this